x-pack/plugins/monitoring/public/components/logstash/pipeline_viewer/views/config_viewer/collapsible_statement.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  EuiButtonEmpty,
+  EuiButtonIcon,
+  EuiCodeBlock,
+  EuiFlexGroup,
+  EuiFlexItem
+} from '@elastic/eui';
+
+function renderStatementName(name, onVertexSelected) {
+  return (
+    <EuiFlexItem
+      grow={false}
+      key="statementName"
+    >
+      <EuiButtonEmpty
+        color="text"
+        size="xs"
+        onClick={onVertexSelected}
+        flush="left"
+      >
+        <span className="configViewer__conditional">{name}</span>
+      </EuiButtonEmpty>
+    </EuiFlexItem>
+  );
+}
+
+function renderIfStatement({ condition }, onVertexSelected) {
+  return [
+    renderStatementName('if', onVertexSelected),
+    (
+      <EuiFlexItem
+        key="ifContent"
+        grow={false}
+      >
+        <EuiCodeBlock
+          fontSize="s"
+          paddingSize="none"
+          transparentBackground={true}
+        >
+          {condition}
+        </EuiCodeBlock>
+      </EuiFlexItem>
+    )
+  ];
+}
+
+function getStatementBody({
+  isIf,
+  statement,
+  statement: { vertex },
+  onShowVertexDetails
+}) {
+  const showVertexDetailsClicked = () => { onShowVertexDetails(vertex); };
+
+  return isIf
+    ? renderIfStatement(statement, showVertexDetailsClicked)
+    : renderStatementName('else', showVertexDetailsClicked);
+}
+
+function getToggleIconType(isCollapsed) {
+  return isCollapsed ? 'arrowRight' : 'arrowDown';
+}
+
+export function CollapsibleStatement(props) {
+  const {
+    collapse,
+    expand,
+    id,
+    isCollapsed
+  } = props;
+
+  const toggleClicked = () => {
+    if (isCollapsed) {
+      expand(id);
+    } else {
+      collapse(id);
+    }
+  };
+
+  return (
+    <EuiFlexGroup
+      responsive={false}
+      gutterSize="none"
+      alignItems="center"
+      className="configViewer__statement"
+    >
+      <EuiFlexItem
+        key={id}
+        grow={false}
+      >
+        <EuiButtonIcon
+          aria-label
+          color="text"
+          iconType={getToggleIconType(isCollapsed)}
+          onClick={toggleClicked}
+          size="s"
+        />
+      </EuiFlexItem>
+      {getStatementBody(props)}
```

Comment:
This is more of a nice to have, so feel free to push it off to a future PR: It'd be nice if `CollapsibleStatement` was a standalone "wrapper" or "decorator" component for _any_ statement component that was passed into it. Concretely that could mean using `props.children` on this line and rendering `CollapsibleStatement` like so:

```js
<CollapsibleStatement ...>
   <IfStatement .../>
</CollapsibleStatement>
```

or

```js
<CollapsibleStatement ...>
   <ElseStatement .../>
</CollapsibleStatement>
```

Reply:
I'd prefer to do this in a separate PR so we can get these existing changes merged faster, especially since there aren't any functional ramifications.

It's tracked in https://github.com/elastic/kibana/issues/19844 and I will update the PR's description to include that in the set of follow-up work.